compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -15810,27 +15827,13 @@ OMR::Z::TreeEvaluator::arraycmpSIMDHelper(TR::Node *node,
 
    // ed : perf : replace these 2 with BRXLE
    //Branch if the number of bytes left is not negative
-   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK3, node, cFlowRegionStart);
-
-   // In certain cases we can branch directly so we must set up the correct global register dependencies
-   TR::RegisterDependencyConditions* compareTargetRDC = getGLRegDepsDependenciesFromIfNode(cg, ificmpNode);
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK3, node, loopStart);
```

Comment:
@r30shah, I observed some strange behaviour here. in some test cases I observed that it jumps even when borrowing happens (lastByteIndexReg is less that 16). The issue happens both with or without my changes. It might be because of my test setup.

Reply:
Thanks @hzongaro. Yes it was because of a known issue explained [here](https://github.com/eclipse/omr/pull/7228#issuecomment-1934923054). I resolve this comment.